File: Core/Models/Stock.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace POS.Core.Models
{
    [Table("Stock")]
    public class Stock : BaseEntity
    {
        [StringLength(35)]
        public string ProductId { get; set; }

        [Required]
        public int Qty { get; set; }

        [ForeignKey("ProductId")]
        public Product Product { get; set; }
    }
}

File: api-server/app/models/stock.go
package models

type Stock struct {
	ID  string `json:"productId" gorm:"primary_key;not null" validate:"required" sql:"type:varchar REFERENCES products(id)"`
	Qty int64  `json:"qty" gorm:"not null" validate:"required"`
	Base

	// Navigation properties
	Product Product `json:"-" gorm:"foreign_key:ProductID" validate:"-"`
}
